File: 2582-pass-the-pillow/2582-pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        direction = time // (n-1)
        additional_people = time % (n-1)
        
        if direction % 2 == 0:
            return 1 + additional_people
        return n - additional_people

File: 2582-pass-the-pillow/2582-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
        int direction = time / (n-1);
        int additionalPeople = time % (n-1);
        
        if (direction % 2 == 0)
            return 1 + additionalPeople;
        return n - additionalPeople;
    }
}